presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/Rule.java

```diff
@@ -21,5 +21,14 @@
 
 public interface Rule
 {
+    /**
+     * Returns a pattern to which plan nodes this rule applies.
+     * Notice that rule may be still invoked for plan nodes which given pattern does not apply,
```

Comment:
yes. but IMO it is design decision which I do not want to take within this PR. See https://github.com/prestodb/presto/pull/7355/commits/6f1bde2162f7c947ac22af2fa6aac8257ebb5f20#r100985176